Firebase/Core/FIRApp.m

```diff
@@ -274,13 +280,15 @@ + (void)addAppToAppDictionary:(FIRApp *)app {
   }
   if ([app configureCore]) {
     sAllApps[app.name] = app;
-    [[NSNotificationCenter defaultCenter]
-        postNotificationName:kFIRAppDiagnosticsNotification
-                      object:nil
-                    userInfo:@{
-                      kFIRAppDiagnosticsConfigurationTypeKey : @(FIRConfigTypeCore),
-                      kFIRAppDiagnosticsFIRAppKey : app
-                    }];
+    if ([app isDataCollectionDefaultEnabled]) {
```

Comment:
It seems like this should only be done in configureCore - but that could be a separate refactor?

Reply:
I'll try to refactor it now.

Reply:
Done.